addons/payment_authorize/models/payment.py

```diff
@@ -188,6 +194,19 @@ def authorize_test_credentials(self):
         transaction = AuthorizeAPI(self.acquirer_id)
         return transaction.test_authenticate()
 
+    @api.multi
+    def write(self, values):
+        super(PaymentAcquirerAuthorize, self).write(values)
+        self._check_authorize_payment_flow()
```

Comment:
```suggestion
        res = super(PaymentAcquirerAuthorize, self).write(values)
        # force the constraint check for any update on acquirers to alert
        # as soon as possible if the acquirer cannot work in its current state
        self._check_authorize_payment_flow()
        return res
```

* write calls return a boolean.  It doesn't mean anything atm, but API should be consistent anyway.
* we can avoid the manual call to the constraint when it was already called by the orm.

Reply:
Indeed, much better. Thank you for your insight